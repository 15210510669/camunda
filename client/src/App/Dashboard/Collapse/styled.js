/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import styled from 'styled-components';

import BasicExpandButton from 'modules/components/ExpandButton';

export const Collapse = styled.div`
  position: relative;
`;

export const ExpandButton = styled(BasicExpandButton)`
  position: absolute;
  top: 14px;
  left: 0;
`;
