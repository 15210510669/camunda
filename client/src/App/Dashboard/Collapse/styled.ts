/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import styled from 'styled-components';

import BasicExpandButton from 'modules/components/ExpandButton';

const Container = styled.div`
  position: relative;
`;

const ExpandButton = styled(BasicExpandButton)`
  position: absolute;
  top: 14px;
  left: 0;
`;

export {Container, ExpandButton};
