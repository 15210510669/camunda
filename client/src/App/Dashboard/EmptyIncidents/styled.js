import styled from 'styled-components';
import {Colors, themed, themeStyle} from 'modules/theme';
import {ReactComponent as Check} from 'modules/components/Icon/check.svg';
import {ReactComponent as Warning} from 'modules/components/Icon/warning-message-icon.svg';

export const EmptyIncidents = themed(styled.div`
  position: absolute;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  border-width: 1px;
  border-style: solid;
  border-color: ${themeStyle({
    dark: Colors.uiDark04,
    light: Colors.uiLight05
  })};
`);

export const Label = themed(styled.span`
  color: ${({type}) => {
    return type === 'success' ? Colors.allIsWell : Colors.incidentsAndErrors;
  }};
  opacity: 0.9;
  font-family: IBMPlexSans;
  font-size: 16px;
`);

export const CheckIcon = styled(Check)`
  width: 18px;
  height: 14px;
  fill: ${Colors.allIsWell};
  margin-right: 13px;
`;
export const WarningIcon = styled(Warning)`
  width: 20px;
  height: 18px;
  fill: ${Colors.incidentsAndErrors};
  margin-right: 15px;
`;
