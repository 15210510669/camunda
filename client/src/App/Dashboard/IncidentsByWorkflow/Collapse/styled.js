import styled from 'styled-components';
import {Colors, themed, themeStyle} from 'modules/theme';
import {ReactComponent as Right} from 'modules/components/Icon/right.svg';
import withStrippedProps from 'modules/utils/withStrippedProps';

export const Collapse = styled.div`
  position: relative;
`;

export const Button = themed(styled.button`
  position: absolute;
  top: 6px;
  left: 0;

  width: 16px;
  height: 16px;
  background: ${themeStyle({
    dark: Colors.uiDark04,
    light: Colors.uiLight03
  })};
  border-radius: 50%;
`);

export const Icon = themed(styled(withStrippedProps(['rotated'])(Right))`
  width: 16px;
  height: 16px;
  position: relative;
  left: -7px;
  top: -1px;

  display: flex;
  align-content: center;
  align-items: center;

  fill: ${themeStyle({
    dark: 'rgba(255, 255, 255, 0.9)',
    light: '#FFFFFF'
  })};

  transform: ${({rotated}) => {
    return rotated ? 'none' : 'rotate(90deg)';
  }};
`);
