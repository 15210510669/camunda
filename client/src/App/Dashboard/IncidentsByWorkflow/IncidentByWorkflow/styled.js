/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import styled, {css} from 'styled-components';
import {Colors, themed, themeStyle} from 'modules/theme';
import withStrippedProps from 'modules/utils/withStrippedProps';

export const Wrapper = themed(styled(withStrippedProps(['perUnit'])('div'))`
  display: flex;
  padding: 0;

  color: ${themeStyle({
    dark: '#fff',
    light: Colors.uiLight06
  })};

  font-family: IBMPlexSans;
  font-size: ${({perUnit}) => (perUnit ? '13px' : '14px')};
  font-weight: ${({perUnit}) => (perUnit ? '400' : '600')}};
  line-height: 1.71;
`);

const redTextStyle = css`
  color: ${Colors.incidentsAndErrors};
`;

const greyTextStyle = css`
  color: ${themeStyle({
    dark: '#ffffff',
    light: Colors.badge02
  })};

  opacity: 0.55;
`;

const greenTextStyle = css`
  color: ${Colors.allIsWell};
  opacity: 0.8;
`;

export const IncidentsCount = themed(styled.div`
  width: 96px;

  ${props => (props.hasIncidents ? redTextStyle : greyTextStyle)}
`);

export const ActiveCount = themed(styled.div`
  margin-left: auto;
  width: 139px;
  text-align: right;

  ${props => (props.hasActive ? greenTextStyle : greyTextStyle)}
`);

export const Label = themed(styled.div`
  opacity: ${themeStyle({
    dark: 0.9,
    light: 1
  })};
`);

const greenBarStyle = css`
  background: ${Colors.allIsWell};
  opacity: 0.8;
`;

const greyBarStyle = css`
  background: ${Colors.badge02};

  opacity: ${themeStyle({
    dark: 0.9,
    light: 0.4
  })};
`;

export const BarContainer = styled.div`
  position: relative;
  > div {
    height: ${props => (props.perUnit ? '3px' : '5px')};
  }
`;

export const Bar = themed(styled.div`
  ${props => (props.hasActive ? greenBarStyle : greyBarStyle)}
`);

export const IncidentsBar = styled.div`
  position: absolute;
  top: 0;

  background: ${Colors.incidentsAndErrors};
`;
