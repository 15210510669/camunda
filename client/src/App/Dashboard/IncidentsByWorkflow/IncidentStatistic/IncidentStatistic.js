import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import * as Styled from './styled.js';

function IncidentStatistic(props) {
  const {label, activeCount, incidentsCount} = props;
  const incidentsBarRatio =
    (100 * incidentsCount) / (activeCount + incidentsCount);
  return (
    <Fragment>
      <Styled.Wrapper>
        <Styled.IncidentsCount>{incidentsCount}</Styled.IncidentsCount>
        <Styled.Label>{label}</Styled.Label>
        <Styled.ActiveCount>{activeCount}</Styled.ActiveCount>
      </Styled.Wrapper>
      <Styled.IncidentStatisticBar>
        <Styled.IncidentsBar
          style={{
            width: `${incidentsBarRatio}%`
          }}
        />
      </Styled.IncidentStatisticBar>
    </Fragment>
  );
}

IncidentStatistic.propTypes = {
  label: PropTypes.string.isRequired,
  activeCount: PropTypes.number.isRequired,
  incidentsCount: PropTypes.number.isRequired
};

export default IncidentStatistic;
