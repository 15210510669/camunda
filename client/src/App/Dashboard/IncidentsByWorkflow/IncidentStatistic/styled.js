import styled from 'styled-components';
import {Colors, themed, themeStyle} from 'modules/theme';
import withStrippedProps from 'modules/utils/withStrippedProps';

export const Wrapper = themed(styled(withStrippedProps(['perUnit'])('div'))`
  display: flex;
  padding: 0;

  color: ${themeStyle({
    dark: '#fff',
    light: Colors.uiLight06
  })};
  opacity: ${themeStyle({
    dark: 0.9,
    light: 1
  })};

  font-family: IBMPlexSans;
  font-size: 14px;
  font-weight: ${({perUnit}) => {
    return perUnit ? '400' : '600';
  }};
  line-height: 1.71;
`);

export const IncidentsCount = styled.div`
  width: 96px;
  color: ${Colors.incidentsAndErrors};
`;

export const ActiveCount = styled.div`
  margin-left: auto;
  width: 139px;
  text-align: right;

  color: ${Colors.allIsWell};
  opacity: 0.8;
`;

export const IncidentStatisticBar = styled.div`
  display: flex;
  height: 3px;
  align-items: stretch;
  background: ${Colors.allIsWell};
  opacity: 0.8;
`;

export const IncidentsBar = styled.div`
  min-width: 1px;

  background: ${Colors.incidentsAndErrors};
`;
