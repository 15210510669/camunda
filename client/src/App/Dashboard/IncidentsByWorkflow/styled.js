import styled from 'styled-components';
import {default as IncidentStatisticComponent} from './IncidentStatistic';

export const Ul = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
`;
export const Li = styled.li`
  margin: 0 0 29px;
  padding: 0;
`;
export const VersionLi = styled.li`
  margin: 21px 0 0;
  padding: 0;
`;

export const IncidentStatistic = styled(IncidentStatisticComponent)`
  margin-left: 32px;
`;
