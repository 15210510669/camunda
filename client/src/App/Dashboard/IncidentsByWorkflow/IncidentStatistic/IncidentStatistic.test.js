import React from 'react';
import {shallow} from 'enzyme';
import IncidentStatistic from './IncidentStatistic';
import * as Styled from './styled';

describe('IncidentStatistic', () => {
  it('should display the right data', () => {
    const node = shallow(
      <IncidentStatistic
        incidentsCount={10}
        label="someLabel"
        activeCount={8}
      />
    );

    expect(node.find(Styled.IncidentsCount).text()).toBe('10');
    expect(node.find(Styled.Label).text()).toBe('someLabel');
    expect(node.find(Styled.ActiveCount).text()).toBe('8');
    expect(node.find(Styled.IncidentsBar)).toExist();
    expect(node).toMatchSnapshot();
  });
});
