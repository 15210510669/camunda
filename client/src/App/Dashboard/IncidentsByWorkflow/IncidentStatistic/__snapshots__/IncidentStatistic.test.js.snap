// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncidentStatistic should display the right data 1`] = `
<div>
  <Themed(Styled(withStrippedProps(Component)))>
    <styled.div>
      10
    </styled.div>
    <div
      data-test="incident-statistic-label"
    >
      someLabel
    </div>
    <styled.div>
      8
    </styled.div>
  </Themed(Styled(withStrippedProps(Component)))>
  <styled.div>
    <styled.div
      style={
        Object {
          "width": "55.55555555555556%",
        }
      }
    />
  </styled.div>
</div>
`;
